content/concepts/introduction/intro-to-consensus.md

```diff
@@ -0,0 +1,63 @@
+# Introduction to Consensus
+
+_Consensus_ is the most important property of any decentralized payment system. In traditional centralized payment systems, one authoritative administrator gets the final say in how and when payments occur. Decentralized systems, by definition, don't have an operator to do that. Instead, decentralized systems like the XRP Ledger define a set of rules all participants follow, so every participant can agree on the exact same series of events and their outcome at any point in time. We call this set of rules a _consensus protocol_.
+
+
+## Consensus Protocol Properties
+
+The XRP Ledger uses a consensus protocol unlike any digital asset that came before it. This protocol, known as the XRP Ledger Consensus Protocol, is designed to have the following important properties:
+
+- Everyone who uses the XRP Ledger can agree on the latest state, and which transactions have occurred in which order.
+- All valid transactions are processed without needing a central operator or having a single point of failure.
+- The ledger can make progress even if some participants join, leave, or behave inappropriately.
+- If too many participants are unreachable or misbehaving, the network fails to make progress rather than diverging or confirming invalid transactions.
+- Confirming transactions does not require wasteful or competitive use of resources, unlike most blockchain systems.
+
+This protocol is still evolving, as is our knowledge of its limits and possible failure cases. For academic research on the protocol itself, see [Consensus Research](consensus-research.html).
+
+## Background
+
+Consensus protocols are a solution to the _double-spend problem_: the challenge to prevent someone from successfully spending the same digital money in two places. The hardest part about this problem is putting transactions in order: without a central authority, it can be difficult to resolve disputes about which transaction comes first when you have two or more mutually-exclusive transactions sent around the same time. For a detailed analysis of the double-spend problem, how the XRP Ledger Consensus Protocol solves this problem, and the tradeoffs and limitations involved, see [Consensus Principles and Rules](consensus-principles-and-rules.html).
+
+
+## Ledger History
+
+The XRP Ledger processes transactions in blocks called "ledger versions", or "ledgers" for short. Each ledger version contains three pieces:
+
+- The current state of all balances and objects stored in the ledger.
+- The set of transactions applied to the previous ledger to result in this one.
```

Comment:
I think I'm a little confused by this point. What does "applied to the previous ledger" mean? Are applied transactions the transactions that are currently being reviewed by the consensus protocol? They've made it into a ledger, but the ledger has not been validated yet?

I think I also get confused because I think of the protocol as working on the current ledger -- when in reality, it sounds like it is working on the previous ledger?

Reply:
State of ledger(n-1) + transactions of ledger(n) = state of ledger(n).

Reply:
In my head, it looks like this pseudocode, but feel free to correct me.

```
Ledger {
  State,
  Transactions,
  Metadata
}

Ledger[n].State = Ledger[n-1].apply(Ledger[n].Transactions)
```

Reply:
Yeah, where the "Metadata" portion tells you exactly how the state has to be modified and the "Transactions" portion explains why.